packages/services/src/services/Translation/__tests__/Translation.test.js

```diff
@@ -131,12 +143,38 @@ describe('TranslationAPI', () => {
     // should contain timestamp
     expect(response).toHaveProperty('timestamp');
 
-    // should not equal old timestamp
-    expect(previousSession.timestamp).not.toEqual(response.timestamp);
+    // should equal mock timestamp
+    expect(response.timestamp).toEqual(mockDate.valueOf());
+
+    // id should have changed
+    expect(response.id).not.toEqual(previousSession.id);
+    expect(response.id).toEqual('TRANSLATION_FRESH');
 
     // timestamps should have at least a two hour difference
     const timeDiff = response.timestamp - previousSession.timestamp,
       _twoHours = 60 * 60 * 2000;
     expect(timeDiff).toBeGreaterThan(_twoHours);
   });
+
+  it('timestamp should not change if within two hours', async () => {
+    // using recent cached session
+    sessionStorageMock.setItem(
+      'dds-translation-us-en',
+      JSON.stringify(responseSuccess)
+    );
+
+    const previousSession = JSON.parse(
+      sessionStorageMock.getItem('dds-translation-us-en')
+    );
+
+    // reinitializing import
+    const TranslationAPI = (await import('../Translation')).default;
+    const response = await TranslationAPI.getTranslation({
+      lc: 'en',
+      cc: 'us',
+    });
+
+    // timestamp should remain unchanged
+    expect(response.timestamp).toEqual(previousSession.timestamp);
+  });
```

Comment:
Do you have some details on what this case tests?

Reply:
We use `responseSuccess` that was just recently refreshed as the cached data. Since it's very recent (just ran), then running the API again shouldn't change the cached data again -- it simply retrieves it. Here we just check if the timestamp remains the same from the previous session.

Reply:
I see, given we don't (and won't) change the timestamp of existing cache, probably we wanted to simply see if the unique ID of `response` equals to one of `previousSession`?